### PR TITLE
WIP, depends on 3285 and 3295

### PR DESCRIPTION
**Pull request by asmacdo**
This is a wishful implementation using the new context managers:
https://pulp.plan.io/issues/3285
https://pulp.plan.io/issues/3295

### CODE REVIEW

**Comment by jortel** [comment]
Clean!

File: pulp_plugin_template/app/tasks.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    #     # Use the WorkingDirectory context manager to set the CWD before writing any files.
+    #     with WorkingDirectory():
+    #         download_and_write_metadata()
+    #         changeset = ChangeSet(importer, created_version, additions=additions, removals=removals)
+    #         changeset.apply_and_drain()
+    raise NotImplementedError
+
+
+@shared_task(base=UserFacingTask)
+def publish(publisher_pk):
```
Comment (dralley) [inline pulp_plugin_template/app/tasks.py]: method header needs to reflect the docstring - missing ```repository_pk```

**Comment by dralley** [comment]
^

**Comment by dralley** [comment]
@asmacdo  I assume this issue should be closed?

**Comment by asmacdo** [comment]
@dralley, yup, thank you.

**Pull request closed without merging**